examples/DisplayTraits/Sample/KittenNode.m

```diff
@@ -0,0 +1,170 @@
+  [self addSubnode:_imageNode];
+  
+  // lorem ipsum text, plus some nice styling
+  _textNode = [[ASTextNode alloc] init];
+  _textNode.attributedString = [[NSAttributedString alloc] initWithString:[self kittyIpsum]
+                                                               attributes:[self textStyle]];
+  _textNode.flexShrink = YES;
+  _textNode.flexGrow = YES;
+  [self addSubnode:_textNode];
+  
+  return self;
+}
+
+- (void)imageTapped:(id)sender
+{
+  if (self.imageTappedBlock) {
+    self.imageTappedBlock();
+  }
+}
+
+- (NSString *)kittyIpsum
+{
+  NSArray *placeholders = [KittenNode placeholders];
+  u_int32_t ipsumCount = (u_int32_t)[placeholders count];
+  u_int32_t location = arc4random_uniform(ipsumCount);
+  u_int32_t length = arc4random_uniform(ipsumCount - location);
+  
+  NSMutableString *string = [placeholders[location] mutableCopy];
+  for (u_int32_t i = location + 1; i < location + length; i++) {
+    [string appendString:(i % 2 == 0) ? @"\n" : @"  "];
+    [string appendString:placeholders[i]];
+  }
+  
+  return string;
+}
+
+- (NSDictionary *)textStyle
+{
+  UIFont *font = [UIFont fontWithName:@"HelveticaNeue" size:12.0f];
+  
+  NSMutableParagraphStyle *style = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
+  style.paragraphSpacing = 0.5 * font.lineHeight;
+  style.hyphenationFactor = 1.0;
+  
+  return @{ NSFontAttributeName: font,
+            NSParagraphStyleAttributeName: style,
+            ASTextNodeWordKerningAttributeName : @.5};
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  
+  ASDisplayTraits displayTraits = self.environmentState.displayTraits;
+  
+  ASStackLayoutSpec *stackSpec = [[ASStackLayoutSpec alloc] init];
+  stackSpec.spacing = kInnerPadding;
+  stackSpec.children = @[_imageNode, _textNode];
+  
+  if (displayTraits.horizontalSizeClass == UIUserInterfaceSizeClassRegular) {
+    _imageNode.alignSelf = ASStackLayoutAlignSelfStart;
+    stackSpec.direction = ASStackLayoutDirectionHorizontal;
+  } else {
+    _imageNode.alignSelf = ASStackLayoutAlignSelfCenter;
+    stackSpec.direction = ASStackLayoutDirectionVertical;
+  }
+  
+  return [ASInsetLayoutSpec insetLayoutSpecWithInsets:UIEdgeInsetsMake(kOuterPadding, kOuterPadding, kOuterPadding, kOuterPadding) child:stackSpec];
+}
+
+
++ (void)defaultImageTappedAction:(ASViewController *)sourceViewController
+{
+  OverrideViewController *overrideVC = [[OverrideViewController alloc] init];
+  
+  overrideVC.overrideDisplayTraitsWithTraitCollection = ^(UITraitCollection *traitCollection) {
+    return (ASDisplayTraits) {
+      .displayScale = traitCollection.displayScale,
+      .horizontalSizeClass = UIUserInterfaceSizeClassCompact,
+      .userInterfaceIdiom = traitCollection.userInterfaceIdiom,
+      .verticalSizeClass = UIUserInterfaceSizeClassCompact,
+      .forceTouchCapability = traitCollection.forceTouchCapability,
+    };
+  };
```

Comment:
Same for here. Seems pretty difficult for API consumer that are not so familiar with structs or so to use the API. Why about creating an object that basically is a representation for a ASDisplayTraits struct and that will be returned in that block. In ASDK if you call the block we use that object and create a ASDisplayTrait for it. Thoughts?

cc @appleguy